### ISSUE TITLE
Supervisor self updater in tight loop if service crashes and fails to start

### ISSUE THREAD

**Issue by chefsalim** [habitat-sh/habitat#7367]
If a service panics on start, the self-updater also restarts. This can cause a tight loop of network traffic. On networks with low bandwidth connectivity or other restrictions, it can cause unrelated issues. This item is to investigate whether/what we can do to manage misbehaved/panicing services.

```
Jan 24 15:19:35 XXXX hab[6911]: hab-launch(SV): Child for service 'YYYY.default' with PID 14347 exited with code exit code: 1
Jan 24 15:19:36 XXXX hab[6911]: hab-sup(SU): Updating from XXXX/YYYY/0.0.9/20200122165940 to XXXX/YYYY/0.0.9/20200124121706
Jan 24 15:19:36 XXXX hab[6911]: thread 'XXXX.default' panicked at 'Main thread has gone away!: "SendError(..)"', src/libcore/result.rs:1084:5
Jan 24 15:19:37 XXXX hab[6911]: YYYY.default(SR): Health checking has been stopped
Jan 24 15:19:37 XXXX hab[6911]: hab-sup(SU): Updating from XXXX/YYYY/0.0.9/20200122165940 to XXXX/YYYY/0.0.9/20200124121706
Jan 24 15:19:37 XXXX hab[6911]: thread 'XXXX.default' panicked at 'Main thread has gone away!: "SendError(..)"', src/libcore/result.rs:1084:5
```

**Comment by christophermaier** [comment]
At this point, I think the best we can do is stop outputting the "scary" panic message, since it's not really a problem, per se, and is generally how this particular code is supposed to behave.

**Comment by christophermaier** [comment]
This is partly addressed with #7390, and should be completely addressed with #7389. I'm closing in favor of those.
